final_project/final_projectApp.swift

```diff
@@ -6,9 +6,22 @@
 //
 
 import SwiftUI
+import FirebaseCore
+import FirebaseAuth
+
+class AppDelegate: NSObject, UIApplicationDelegate {
+  func application(_ application: UIApplication,
+                   didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
+    FirebaseApp.configure()
+
+    return true
+  }
+}
 
 @main
 struct final_projectApp: App {
+    @UIApplicationDelegateAdaptor(AppDelegate.self) var delegate
+    
```

Comment:
You can initialize the @StateObject for your viewModel here and pass the value to ConteteView as an environment object.

> @StateObject var viewModel = LoginViewModel()
>    var body: some Scene {
>        WindowGroup {
>            ContentView()
>                .environmentObject(viewModel)
>        }
    }

Reply:
Got it this makes sense!